### ISSUE TITLE
Propagate labels from Pipeline and Task to PipelineRun and TaskRun

### ISSUE THREAD

**Issue by dwnusbaum** [knative/build-pipeline#501]
## Expected Behavior

Custom labels specified on `Pipeline` and `Task` resources should be added to `PipelineRun` and `TaskRun` objects that reference them. In other words, labels should propagate as follows:

Pipeline -> PipelineRun -> TaskRun -> Pod
Task -> TaskRun -> Pod

In addition, similarly to how the `pipeline.knative.dev/pipeline` label is added to generated `TaskRuns`, a `pipeline.knative.dev/task` label should be added to generated `Pods`.

## Actual Behavior

Custom labels on `Pipeline` and `Task` resources are not propagated to other resources. Labels propagate as follows:

PipelineRun -> TaskRun -> Pod
TaskRun -> Pod

## Additional Info

See #458. #488 added label propagation from `PipelineRun` to `TaskRun` to `Pod`, but did not handle propagation for `Pipeline` and `Task`. Propagating `Pipeline` and `Task` labels will require updating a few data structures so that either the `ObjectMeta` field or the labels field specifically of `Pipelines` and `Tasks` is available when `TaskRuns` and `Pods` are created so that they can be propagated correctly.

Consistent label propagation is desirable because it allows users to easily find all resources associated with a given Pipeline/Task.

Open questions:

* Should labels be propagated from/to resources as well?

**Comment by dwnusbaum** [comment]
Any thoughts on the best way to modify `ResolvedTaskResources` so it contains task metadata as well? I see two obvious options, but I'm not sure if there is a particular reason to prefer either:

1. Replace `TaskSpec *v1alpha1.TaskSpec` with `Task v1alpha1.TaskInterface`
2. Add a new field `TaskMetadata *metav1.ObjectMeta` (or just `TaskLabels map[string]string` if we don't want all of the metadata)

**Comment by dwnusbaum** [comment]
Turns out that I was a bit confused, and I don't think we ended up needing to change any data types. I think the main question left is how best to update the PipelineRun/TaskRun with the labels from the Pipeline/Task so that we don't keep doing the same work in the reconciler and so we can keep the core status logic clear. See https://github.com/knative/build-pipeline/pull/517#discussion_r257839853.
